### PR TITLE
New. Integrations. Added protect for outside iframe.

### PR DESCRIPTION
**Pull request by svfcode**
task https://doboard.com/1/task/5607

### CODE REVIEW

File: js/src/apbct-public--5--external-forms.js [hunk 1]
```diff
@@ -292,9 +292,81 @@ window.onload = function() {
...
+    cover.style.top = 0;
+    cover.onclick = function(e) {
+        if (ctProtectOutsideIframeCheck === undefined) {
+            let currentDiv = e.currentTarget;
+            currentDiv.style.opacity = 0.5;
+            let preloader = document.createElement('div');
+            preloader.className = 'apbct-iframe-preloader';
+            currentDiv.appendChild(preloader);
+
+            let data = {
```
Comment (alexandergull) [inline js/src/apbct-public--5--external-forms.js:337]: What about event_token? We should check if event_token is available on the page and add it to the data if so. 
```
            const token = apbctLocalStorage.get('bot_detector_event_token');
```

Also, we must generate event_token somehow for this page if no forms available  besides the iframe

File: js/src/apbct-public--5--external-forms.js [hunk 2]
```diff
@@ -292,9 +292,81 @@ window.onload = function() {
...
+    }
+}
+
+let ctProtectOutsideIframeCheck;
+/**
+ * Protect forms placed in iframe with outside src handler
+ * @param {HTMLElement} iframe
+ */
+function ctProtectOutsideIframeHandler(iframe) {
+    let cover = document.createElement('div');
```
Comment (alexandergull) [inline js/src/apbct-public--5--external-forms.js:322]: div with no id is bad practice, isnt it?

**Review by Glomberg — APPROVED** [review]

File: js/src/apbct-public--5--external-forms.js (hunk 2 above)
Comment (svfcode) [reply js/src/apbct-public--5--external-forms.js:323]: I think it is normal.

File: js/src/apbct-public--5--external-forms.js (hunk 1 above)
Comment (svfcode) [reply js/src/apbct-public--5--external-forms.js:342]: The bot detector carries additional information only if there are additional forms on the page. Therefore, it makes no sense to generate it if there are no other forms.

Comment (svfcode) [reply js/src/apbct-public--5--external-forms.js:342]: So, if another form exists added event token to request body https://github.com/CleanTalk/wordpress-antispam/pull/335/commits/1fc2dd0c12276785b65150edb4846f6ca94f6935

**Pull request merged**